core/services/synchronization/telemetry_ingress_client.go

```diff
@@ -211,11 +192,18 @@ func (tc *telemetryIngressClient) getCSAPrivateKey() (privkey []byte, err error)
 // Send sends telemetry to the ingress server using wsrpc if the client is ready.
 // Also stores telemetry in a small buffer in case of backpressure from wsrpc,
 // throwing away messages once buffer is full
-func (tc *telemetryIngressClient) Send(payload TelemPayload) {
+func (tc *telemetryIngressClient) Send(ctx context.Context, telemData []byte, contractID string, telemType TelemetryType) {
+	payload := TelemPayload{
+		Ctx:        context.Background(),
```

Comment:
Sorry I missed the thread on this, but typically we want to derive the context from `tc.chDone` in cases like this (we have `utils.StopChan` to simplify things), so we can still cancel everything on shutdown. Otherwise, if this is unusual for some reason then we should leave a comment explaining why,

Reply:
Could the `handleTelemetry()` loop just make one local context to reuse for every call? Then this field could be dropped, which would be more idiomatic anyways.

Reply:
Addresed in [`73fcb44` (#10623)](https://github.com/smartcontractkit/chainlink/pull/10623/commits/73fcb4493c7c6cefc17708a35aeb6ac38665cd67)